lib/Optimizer/Transforms/RegToMem.cpp

```diff
@@ -157,22 +167,32 @@ struct RegToMemAnalysis {
       }
     });
     func.walk([&](Operation *op) {
-      // FIXME: This really needs to figure out which operands are wires and
-      // which are not. It blindly assumes everything is a wire.
       if (isa<RAW_MEASURE_OPS>(op)) {
-        for (auto [t, r] :
-             llvm::zip(op->getOperands(), op->getResults().drop_front()))
+        auto wireOpnds = collectLinearValues(op->getOperands());
+        for (auto [t, r] : llvm::zip(wireOpnds, op->getResults().drop_front()))
           insertToEqClass(t, r);
       } else if (isa<RAW_GATE_OPS>(op)) {
         auto gate = cast<quake::OperatorInterface>(op);
-        for (auto [t, r] : llvm::zip(gate.getControls(), op->getResults()))
+        // Check that the IR is not in the pruned control form.
+        if (std::any_of(
+                gate.getControls().begin(), gate.getControls().end(),
+                [](Value v) { return isa<quake::ControlType>(v.getType()); })) {
```

Comment:
I don't care if you change this, but just FYI this could use `llvm::any_of` in order to avoid providing both the .begin() and .end() parameters.